proxy-node-translator/src/main/java/uk/gov/ida/notification/translator/saml/HubResponseContainer.java

```diff
@@ -11,28 +12,43 @@
 
 public class HubResponseContainer {
 
-    private final String pid;
+    private final Optional<String> pid;
     private final String eidasRequestId;
     private final URI destinationURL;
     private final URI issuer;
     private final Attributes attributes;
     private final VspScenario vspScenario;
     private final VspLevelOfAssurance levelOfAssurance;
+    private final boolean eidasTransientPid;
+
 
     public HubResponseContainer(
             final HubResponseTranslatorRequest hubResponseTranslatorRequest,
-            final TranslatedHubResponse translatedHubResponse) {
-        this.pid = translatedHubResponse.getPid().orElse(null);
+            final TranslatedHubResponse translatedHubResponse,
+            final IdentifierGenerationStrategy identifierGenerator) {
+        this.pid = generatePidForNameID(hubResponseTranslatorRequest, translatedHubResponse, identifierGenerator);
```

Comment:
Don't do this here - the PID is assigned a value in `HubResponseTranslator` so no need to do it twice